src/viking.js

```diff
@@ -1,11 +1,90 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health,strength){
+      this.health = health;
+      this.strength = strength;
+    }
+    attack(){
+        return this.strength;
+      };
+    receiveDamage(damage){
+        this.health -= damage;
+      };
+    
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier{
+    constructor(name,health,strength){
+        super(health,strength)
+        this.name = name;
+    }
+    battleCry(){
+            return  "Odin Owns You All!";
+    };
+    receiveDamage (damage){
+            let pointsLife = this.health -= damage;
+            if( pointsLife > damage){
+                return `${this.name} has received ${damage} points of damage`;
+            }else{
+                return `${this.name} has died in act of combat`;
+            }
+        }
+        
+    
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health,strength){
+        super(health,strength)
+    }
+    receiveDamage(damage){
+            let pointsLife = this.health -= damage;
+            if( pointsLife > damage){
+                return `A Saxon has received ${damage} points of damage`;
+            }else{
+                return `A Saxon has died in combat`
+            }
+    }
+    
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+    addViking(viking){
+          this.vikingArmy.push(viking)
+    };
+    addSaxon(saxon){
+          this.saxonArmy.push(saxon)
+    };
+    vikingAttack(){
+            //this.saxonArmy[0].health = Math.floor(Math.random()*saxonArmy.length)
+            //this.vikingArmy[0].strength = Math.floor(Math.random()*vikingArmy.length)
+            if(saxon.health === vikingAmry.strength){
+             splice(0,saxonArmy.length)
+             return "A Saxon has died in combat"
+            }
```

Comment:
No método vikingAttack, vc precisa antes selecionar index do viking e index do saxon, para podermos identificar qual viking estarána luta e qual saxon, tudo isso de maneira aleatória, exemplo:
let randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length)
let randomViking = this.vikingArmy[randomVikingIndex
Agora vc faz a mesma lógica, só que para os saxons. 
Na luta, vc pode colocar numa variável o ataque, mais ou menos assim:
let fight = randomViking.receiveDamage(randomSaxon.attack())
        if(randomViking.health <= 0){
            this.vikingArmy.splice(randomVikingIndex, 1)
        }
        return fight
} terminando aqui o código desse método